classes/profile.php

```diff
@@ -97,16 +97,78 @@ public static function getashar($id): string {
     }
 
     /**
-     * Gets the script type of th request.
+     * @return int
+     * @throws \dml_exception
+     */
+    public static function getnumautoprofiles(): int {
+        global $DB;
+        return $DB->get_field_sql(
+            "SELECT count(*) FROM {tool_excimer_profiles} WHERE logmethod = ?",
+            [ manager::LOGMETHOD_AUTO ]
+        );
+    }
+
+    /**
+     * @return object
+     * @throws \dml_exception
+     */
+    public static function getfastestautoprofile(): object {
+        global $DB;
+        $sql = "
+            SELECT id, duration
+            FROM {tool_excimer_profiles}
+            WHERE logmethod = ?
+            ORDER BY duration ASC limit 1
+        ";
+        return $DB->get_record_sql($sql, [ manager::LOGMETHOD_AUTO ]);
+    }
+
+    /**
+     * @throws \dml_exception
+     */
+    public static function purgefastestautoprofile(): void {
+        global $DB;
+        $sql = "
+            DELETE FROM {tool_excimer_profiles}
+            WHERE id = (
+                SELECT id
+                FROM {tool_excimer_profiles}
+                WHERE logmethod = ?
+                ORDER BY duration ASC LIMIT 1
+            )
+        ";
+        $DB->execute($sql, [ manager::LOGMETHOD_AUTO ]);
+    }
+
+    /**
+     * @param $numtopurge
+     * @throws \dml_exception
+     */
+    public static function purgefastestautoprofiles($numtopurge): void {
+        global $DB;
+        $sql = "
+            DELETE FROM {tool_excimer_profiles}
```

Comment:
please convert this to use one of the dml methods, probably $DB->delete_records_subquery

Reply:
also would it be easier to swap this around so the param is the number you want to keep?

Then purgefastestautoprofile could be redundant as you'd just say purge_fastest_autoprofiles($keep = 0)

Reply:
It is more efficient this way. I can just list them in order and slice from the top. If you prefer to use num-to-keep, I can change it, but it will require extra logic.

Reply:
The function has been converted to use DML method.